iot_packet_verifier/src/verifier.rs

```diff
@@ -174,9 +182,89 @@ pub trait ConfigServer {
     ) -> Result<PublicKeyBinary, Self::Error>;
 
     async fn disable_org(&self, oui: u64) -> Result<(), Self::Error>;
+
+    async fn enable_org(&self, oui: u64) -> Result<(), Self::Error>;
+
+    async fn list_orgs(&self) -> Result<Vec<Org>, Self::Error>;
+
+    async fn monitor_funds<S, B>(
+        self,
+        solana: S,
+        balances: B,
+        minimum_allowed_balance: u64,
+        monitor_period: Duration,
+        shutdown: triggered::Listener,
+    ) -> Result<(), MonitorError<S::Error, Self::Error>>
+    where
+        S: SolanaNetwork,
+        B: BalanceStore,
+    {
+        let join_handle = tokio::spawn(async move {
+            loop {
+                tracing::info!("Checking if any orgs need to be re-enabled");
+
+                for Org { locked, payer, oui } in self
+                    .list_orgs()
+                    .await
+                    .map_err(MonitorError::ConfigClientError)?
+                    .into_iter()
+                {
+                    if locked {
+                        let balance = solana
+                            .payer_balance(&payer)
+                            .await
+                            .map_err(MonitorError::SolanaError)?;
+                        if balance >= minimum_allowed_balance {
+                            balances.set_balance(&payer, balance).await;
+                            self.enable_org(oui)
+                                .await
+                                .map_err(MonitorError::ConfigClientError)?;
+                        }
+                    }
+                }
+                // Sleep until we should re-check the monitor
+                sleep_until(Instant::now() + monitor_period).await;
+            }
+        });
+        tokio::select! {
+            result = join_handle => match result {
+                Ok(Ok(())) => Ok(()),
+                Ok(Err(err)) => Err(err),
+                Err(err) => Err(MonitorError::from(err)),
+            },
+            _ = shutdown => Ok(())
+        }
+    }
 }
 
-// TODO: Move this somewhere else
+#[async_trait]
+pub trait BalanceStore: Send + Sync + 'static {
+    async fn set_balance(&self, payer: &PublicKeyBinary, balance: u64);
+}
+
+#[async_trait]
+impl BalanceStore for crate::balances::BalanceStore {
+    async fn set_balance(&self, payer: &PublicKeyBinary, balance: u64) {
+        self.lock().await.entry(payer.clone()).or_default().balance = balance;
+    }
+}
+
+#[async_trait]
+impl BalanceStore for Arc<Mutex<HashMap<PublicKeyBinary, u64>>> {
```

Comment:
```suggestion
// differs from the BalanceStore in the value stored in the contained HashMap; a u64 here instead of a Balance {} struct
impl BalanceStore for Arc<Mutex<HashMap<PublicKeyBinary, u64>>> {
```
took me a moment to realize why we need this separate trait impl